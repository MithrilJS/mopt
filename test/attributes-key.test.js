"use strict";

var code = require("./lib/code");

describe("Attributes", () => {
    describe("key", () => {
        it("should extract the key attribute (number)", () =>
            expect(
                code(`m("div", { key : 1 })`)
            )
            .toMatchSnapshot()
        );

        it("should extract the key attribute (string)", () =>
            expect(
                code(`m("div", { key : "1" })`)
            )
            .toMatchSnapshot()
        );

        it("should extract the key attribute (boolean)", () =>
            expect(
                code(`m("div", { key : true })`)
            )
            .toMatchSnapshot()
        );

        it("should extract the key attribute (identifier)", () =>
            expect(
                code(`m("div", { key : fooga })`)
            )
            .toMatchSnapshot()
        );
    });
});
