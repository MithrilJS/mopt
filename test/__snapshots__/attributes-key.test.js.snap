// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attributes key should extract the key attribute (boolean) 1`] = `"m.vnode(\\"div\\",true,undefined,[],undefined,undefined);"`;

exports[`Attributes key should extract the key attribute (identifier) 1`] = `"m.vnode(\\"div\\",fooga,undefined,[],undefined,undefined);"`;

exports[`Attributes key should extract the key attribute (number) 1`] = `"m.vnode(\\"div\\",1,undefined,[],undefined,undefined);"`;

exports[`Attributes key should extract the key attribute (string) 1`] = `"m.vnode(\\"div\\",\\"1\\",undefined,[],undefined,undefined);"`;
